### DISCUSSION TITLE
Triggering an edge function or API call on user signup?

### DISCUSSION THREAD

**Discussion by scalemaildev** [supabase/supabase#9016] [Questions] [answered]
I am building a webapp that will require a web3 account to be created for each user, which is then managed by the webapp "under the hood".

I am exploring auth options for this app, and Supabase stood out as a strong potential choice. I am wondering if Supabase has a built-in method for triggering a Javascript/Typescript edge function (or an API call) on signup, so that I can create these web3 accounts and assign them to the users as part of the signup process.

Is there any example of this being done with Supabase? Thanks.

**e: I'm going to ask this in the Discord instead since it seems more active.**

**Comment by Edditoria** [comment, marked as answer]
I try to guess about your use case:

Instead of insert data to DB table directly, I think you may want to call Edge Functions to do a few jobs:

1. `signUp()` for the user.
2. Insert record to a *profile* table, that contains user info, `web3_account_id_or_something` (string or boolean), etc.
3. Call the API to register a web3 account you need, the Edge Function get the result.
4. If the result is success, the Function update the `web3_account_id...` and returns that result to the user.
5. If the result is fail, you may want to use another routine to schedule a job in backend, an d let the user "click to retry".

You may want to build 2 functions to handle registration on your app and web3 accounts. So your user can still explore your application while initiate/update/follow-up/notify for their web3-registration status.

I think the blog post [Using Edge Functions in Supabase: A complete guide](https://blog.logrocket.com/using-edge-functions-supabase-complete-guide/) is pretty good to show you how to initiate, consume an API from 3rd party, and return a result. But I would put more effort on handling fail cases for a better user-experience.

Hopefully I can help a little.
